plugins/com.google.cloud.tools.eclipse.appengine.deploy.ui/src/com/google/cloud/tools/eclipse/appengine/deploy/ui/DeployConsolePageParticipant.java

```diff
@@ -0,0 +1,129 @@
+
+  private DeployConsole console;
+  private Action terminateAction;
+  private Action closeAction;
+
+  @Override
+  public void init(IPageBookViewPage page, IConsole console) {
+    if (console instanceof DeployConsole) {
+      this.console = (DeployConsole) console;
+    } else {
+      logger.log(Level.SEVERE,
+                 "console is instance of {0}, expected was {1}",
+                 new Object[]{ console.getClass().getName(), DeployConsole.class.getName() });
+    }
+    IActionBars actionBars = page.getSite().getActionBars();
+    configureToolBar(actionBars.getToolBarManager());
+    addJobchangeListener();
+    update();
+  }
+
+  private void configureToolBar(IToolBarManager toolbarManager) {
+    terminateAction = createTerminateAction();
+    toolbarManager.appendToGroup(IConsoleConstants.LAUNCH_GROUP, terminateAction);
+
+    closeAction = createCloseAction();
+    toolbarManager.appendToGroup(IConsoleConstants.LAUNCH_GROUP, closeAction);
+  }
+
+  private void addJobchangeListener() {
+    console.getJob().addJobChangeListener(new JobChangeAdapter() {
+      public void done(IJobChangeEvent event) {
+        System.out.println("done update");
+        update();
+      }
+    });
+  }
+
+  private void update() {
+    StandardDeployJob tag = console.getJob();
+
+    if (terminateAction != null) {
+      terminateAction.setEnabled(tag.getState() != Job.NONE);
+    }
+
+    if (closeAction != null) {
+      closeAction.setEnabled(tag.getState() == Job.NONE);
+    }
+  }
+
+  private Action createCloseAction() {
+    Action close = new Action(Messages.getString("action.close")) {
+      @Override
+      public void run() {
+        ConsolePlugin.getDefault().getConsoleManager().removeConsoles(new IConsole[] { console });
+      }
+    };
+    close.setToolTipText(Messages.getString("action.close"));
+    close.setImageDescriptor(getSharedImage(ISharedImages.IMG_ELCL_REMOVE));
+    close.setHoverImageDescriptor(getSharedImage(ISharedImages.IMG_ELCL_REMOVE));
+    close.setDisabledImageDescriptor(getSharedImage(ISharedImages.IMG_ELCL_REMOVE_DISABLED));
+    return close;
+  }
+
+  private Action createTerminateAction() {
+    Action terminate = new Action(Messages.getString("action.stop")) {
+      @Override
+      public void run() {
+        console.getJob().cancel();
+        update();
+      }
+    };
+    terminate.setToolTipText(Messages.getString("action.stop"));
+    terminate.setImageDescriptor(getSharedImage(ISharedImages.IMG_ELCL_STOP));
+    terminate.setHoverImageDescriptor(getSharedImage(ISharedImages.IMG_ELCL_STOP));
+    terminate.setDisabledImageDescriptor(getSharedImage(ISharedImages.IMG_ELCL_STOP_DISABLED));
+    return terminate;
+  }
+
+  private ImageDescriptor getSharedImage(String image) {
+    return PlatformUI.getWorkbench().getSharedImages().getImageDescriptor(image);
+  }
+
+  @Override
+  public void activated() {
```

Comment:
How would that interfere with the enablement of the actions? Not that I'm fighting against adding `update()` to `activate()` just curious, because I'd rather not add code that is redundant.